docs/src/proposals/scrambled-tx.md

```diff
@@ -0,0 +1,335 @@
+validators (This average duration will be the basis of difficulty adjustment of
+scrambling).
+
+Also, rpc can notify optimistic execution signal to clients at this point. This
+signifies subsequent pipeline processing is fully deterministic and not reverted
+unless opt-conf is violated (some portion of stake is slash-able).
+
+At the same time, the validators also initiate the descrambling stage for the
+block in a random, collaborative and distributed manner. Once they solve their
+chosen chunk of scrambled transactions, they send revealed encryption keys to
+the leader to be propagated via turbine likewise the block shreds. Also
+sig-verify is done at the end of descrambling.
+
+After all transactions are descrambled from the cooperative efforts, the block
+can be finally scheduled and executed deterministically.
+
+Lastly, validators votes on the reproduced bank hash after replaying the
+processing likewise. Then, rpc can notify optimistic confirmation signal to
+clients.
+
+### Downsides
+
+- Increased latency
+- Some computational cost on clients
+- Reliance of local wall clock time
+- Pretty invasive system change
+
+
+### Assumptions of other subsystem's improvements
+
+- bankless leader
+- randomized leader schedule (or stop 4-slot grouping at the very least)
+- on-chain mempool with bidding
+- quic adaption for tpu (for ipv6-mtu-exeeding transaction packet and ddos prevention)
+
+### Scrambling details
+
+As said above, scrambling refers to a specially-arranged encryption scheme where
+anyone can decrypt (= reveal) the transaction details after a non-parallelizable
+computation which takes certain amount of time.
+
+On top of it, scrambling also has a nice property where the revelation can
+easily be verified by anyone with some kind of message authentication tag. This
+means rpc node doesn't need GPU.
+
+All of fields in a transaction are scrambled with randomized padding, including
+fee payer to avoid any kind of fingerprinting while allowing garbage-based
+spamming with amplification risk.
+
+Also, scrambled transaction expires very quickly based on wall clock time, less
+than the targeted duration of difficulty to protect from processing stalls by
+malicious leaders for MEV.
+
+So, scrambled transaction binary format will be like this
+
+```
+[scrambled transaction marker]
+[seed to recursive sha256]
+[recursion count]
+aes256-ed..
+  [aes256 key used to scramble] # this works as authentication tag
+  [expiration in unix time]
+  [normal transaction details...] # fee payer sig cover the scrambling key to avoid unauthorized scrambling
+```
+
+So, this assumes quic to bypass the transaction udp packet size limit.
+
+The number of sha256 recursion is chosen so that anyone can't reveal the
+transaction details, until the cluster optimistically commits the ordering of
+transactions in a block with very high certainty. So, this assumes the recursive
+sha256 is exhibits rather uniform throughput across various hardware (cpu/gpu)
```

Comment:
> ... uniform throughput ...

if this isn't true, solana needs another subsystem or node termed like `descrambling pool`...

Reply:
so the solution to this sub probmlem would be something like `temporal capacity-based work spread based on each leader's descrambling reward distribution`.